src/lib.rs

```diff
@@ -188,7 +519,12 @@ impl UTransport for ULinkZenoh {
         let zenoh_key = ULinkZenoh::to_zenoh_key_string(&topic)?;
         // Generate listener string for users to delete
         let mut hashmap_key = format!("{}_{:X}", zenoh_key, rand::random::<u64>());
-        while self.map.lock().unwrap().contains_key(&hashmap_key) {
+        while self
```

Comment:
You could take the `lock()` only once before the `while` loop. Alternatively to generate a random number, you could use a `64bit` counter.

Reply:
That's a wise idea! We don't need to handle the conflict since 64bit is nearly impossible to run out.